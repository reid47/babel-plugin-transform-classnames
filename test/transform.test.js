import { transformSync } from '@babel/core';
import plugin from '../src/index';

const transform = input => {
  return transformSync(input, {plugins: [plugin]}).code;
}

test('string literals', () => {
  const input = `
    import cn from 'classnames.macro';

    cn('wow', 'cool', 'strings');
  `;

  expect(transform(input)).toMatchSnapshot();
});

test('string literals with variable at end', () => {
  const input = `
    import cn from 'classnames.macro';

    cn('wow', 'cool', 'strings', something);
  `;

  expect(transform(input)).toMatchSnapshot();
});

test('string literals with variable at beginning', () => {
  const input = `
    import cn from 'classnames.macro';

    cn(something, 'wow', 'cool', 'strings');
  `;

  expect(transform(input)).toMatchSnapshot();
});

test('string literals with variable in the middle', () => {
  const input = `
    import cn from 'classnames.macro';

    cn('wow', something, 'cool', 'strings');
  `;

  expect(transform(input)).toMatchSnapshot();
});

test('all variables', () => {
  const input = `
    import cn from 'classnames.macro';

    cn(some, things, here);
  `;

  expect(transform(input)).toMatchSnapshot();
});

test('object literal', () => {
  const input = `
    import cn from 'classnames.macro';

    cn({key: maybe});
  `;

  expect(transform(input)).toMatchSnapshot();
});

test('object literal + strings', () => {
  const input = `
    import cn from 'classnames.macro';

    cn('hey', {key: maybe}, 'nice', 'wow');
  `;

  expect(transform(input)).toMatchSnapshot();
});

test('object literal with multiple keys', () => {
  const input = `
    import cn from 'classnames.macro';

    cn({key: maybe, hello: world});
  `;

  expect(transform(input)).toMatchSnapshot();
});

test('object literal with boolean keys', () => {
  const input = `
    import cn from 'classnames.macro';

    cn({always: true, never: false});
  `;

  expect(transform(input)).toMatchSnapshot();
});

test('multiple object literals', () => {
  const input = `
    import cn from 'classnames.macro';

    cn({always: true, never: false}, "string", {hello: world}, {another: obj});
  `;

  expect(transform(input)).toMatchSnapshot();
});