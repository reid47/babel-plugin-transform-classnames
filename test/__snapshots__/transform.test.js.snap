// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all variables 1`] = `"(some || \\"\\") + \\" \\" + (things || \\"\\") + \\" \\" + (here || \\"\\");"`;

exports[`multiple object literals 1`] = `"\\"always string\\" + (world ? \\" hello\\" : \\"\\") + (obj ? \\" another\\" : \\"\\");"`;

exports[`object literal + strings 1`] = `"'hey' + (maybe ? \\" key\\" : \\"\\") + \\" nice wow\\";"`;

exports[`object literal 1`] = `"maybe ? \\" key\\" : \\"\\";"`;

exports[`object literal with boolean keys 1`] = `"\\"always\\";"`;

exports[`object literal with multiple keys 1`] = `"(maybe ? \\" key\\" : \\"\\") + (world ? \\" hello\\" : \\"\\");"`;

exports[`string literals 1`] = `"\\"wow cool strings\\";"`;

exports[`string literals with variable at beginning 1`] = `"(something || \\"\\") + \\" wow cool strings\\";"`;

exports[`string literals with variable at end 1`] = `"\\"wow cool strings \\" + (something || \\"\\");"`;

exports[`string literals with variable in the middle 1`] = `"\\"wow \\" + (something || \\"\\") + \\" cool strings\\";"`;
